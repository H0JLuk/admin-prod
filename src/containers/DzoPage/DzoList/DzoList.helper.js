export const sortDzoListBySearchParams = ({ sortBy, direction, filterText }, copyDzoList) => {
    return (!filterText
        ? copyDzoList.slice()
        : copyDzoList.filter(({ dzoName }) => dzoName.toLowerCase().includes(filterText.toLowerCase()))
    ).sort((a, b) => sortItem(sortBy, direction, [a, b]));
};

function sortItem(type, direction, sortableItems) {
    const [a, b] = sortableItems;

    const ASC_SORT = stringToLowerCase(a[type]) < stringToLowerCase(b[type]) ? -1 : 1;
    const DESC_SORT = stringToLowerCase(b[type]) > stringToLowerCase(a[type]) ? 1 : -1;

    switch (type) {
        case 'dzoId': {
            return direction === 'ASC' ? a[type] - b[type] : b[type] - a[type];
        }
        default: {
            return direction === 'ASC' ? ASC_SORT : DESC_SORT;
        }
    }
}

const stringToLowerCase = (value) => {
    return typeof value === 'string' ? value.toLowerCase() : value;
};