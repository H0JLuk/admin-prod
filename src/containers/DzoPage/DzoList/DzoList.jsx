import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { generatePath, useHistory, useLocation } from 'react-router-dom';
import { Button } from 'antd';
import { deleteDzo, getAllDzoList, getDzoList } from '../../../api/services/dzoService';
import DzoListTable from './Table/DzoListTable';
import Header from '../../../components/Header/Header';
import TableDeleteModal from '../../../components/TableDeleteModal/TableDeleteModal';
import HeaderWithActions from '../../../components/HeaderWithActions/HeaderWithActions';

import { defaultSearchParams, getSearchParamsFromUrl, sortItemsBySearchParams } from '../../../utils/helper';
import { DZO_PAGES } from '../../../constants/route';

import styles from './DzoList.module.css';

const MODAL_TITLE = 'Вы уверены, что хотите удалить эти ДЗО?';
const MODAL_SUCCESS_TITLE = 'Результат удаления ДЗО';

const DZO_TITLE = 'Список ДЗО';

const DROPDOWN_SORT_MENU = [
    { name: 'dzoName', label: 'По названию' },
    { name: 'dzoId', label: 'По ID' },
    { name: 'dzoCode', label: 'По коду' },
];

const BUTTON_TEXT = {
    SAVE: 'Сохранить',
    ADD: 'Добавить',
    CANCEL: 'Отменить',
    SELECT: 'Выбрать',
    SELECT_ALL: 'Выбрать все',
    CANCEL_ALL: 'Отменить все',
    DELETE: 'Удалить',
    CHOSEN: 'Выбрано',
};

const SEARCH_INPUT_PLACEHOLDER = 'Поиск по названию дзо';

const getURLSearchParams = ({ ...rest }) => new URLSearchParams(rest).toString();

const defaultSelected = { rowValues: [], rowKeys: [] };

const DzoPage = ({ matchPath }) => {
    const history = useHistory();
    const { search } = useLocation();
    const [params, setParams] = useState(defaultSearchParams);
    const [loadingTable, setLoadingTable] = useState(true);
    const [dzoList, setDzoList] = useState([]);
    const [copyDzoList, setCopyDzoList] = useState([]);
    const [select, setSelect] = useState(false);
    const [selectedItems, setSelectedItems] = useState(defaultSelected);
    const [isModalView, setIsModalView] = useState(false);
    const dzoCodes = useRef([]);

    const onAddDzo = () => history.push(generatePath(`${matchPath}${DZO_PAGES.ADD_DZO}`), { dzoCodes: dzoCodes.current });

    const loadDzoList = useCallback(async (searchParams = defaultSearchParams) => {
        const urlSearchParams = getURLSearchParams(searchParams);
        setLoadingTable(true);
        try {
            const { dzoDtoList = [] } = await getDzoList();
            const { dzoDtoList: allDzo = [] } = await getAllDzoList();
            dzoCodes.current = allDzo.map(({ dzoCode }) => dzoCode);
            history.replace(`${matchPath}?${urlSearchParams}`);
            clearSelectedItems();
            setParams({ ...searchParams });
            setCopyDzoList(dzoDtoList);
            setDzoList(sortItemsBySearchParams(searchParams, dzoDtoList, 'dzoName'));
        } catch (e) {
            console.warn(e);
        }
        setLoadingTable(false);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    useEffect(() => {
        loadDzoList(getSearchParamsFromUrl(search));
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [loadDzoList]);

    const sortDzoList = useCallback((searchParams = defaultSearchParams) => {
        const { filterText, sortBy } = searchParams;
        setParams(searchParams);
        history.replace(`${matchPath}?${getURLSearchParams(searchParams)}`);

        if (!filterText && !sortBy) {
            setDzoList(copyDzoList);
            return;
        }

        setDzoList(sortItemsBySearchParams(searchParams, copyDzoList, 'dzoName'));
    }, [copyDzoList, history, matchPath]);

    const clearSelectedItems = () => {
        setSelectedItems(defaultSelected);
    };

    const clearSelectsRow = () => {
        setSelect(false);
        clearSelectedItems();
    };

    const updateSelected = useCallback((rowKeys, rowValues) => {
        setSelectedItems({ rowKeys, rowValues });
    }, []);

    const onSelect = useCallback(() => setSelect(true), []);

    const rowSelection = useMemo(() => ({
        selectedRowKeys: selectedItems.rowKeys,
        onChange: updateSelected,
    }), [selectedItems, updateSelected]);

    const selectAll = useCallback(() => {
        if (dzoList.length !== selectedItems.rowKeys.length) {
            setSelectedItems({ rowKeys: dzoList.map(({ dzoId }) => dzoId), rowValues: dzoList });
            return;
        }
        clearSelectedItems();
    }, [dzoList, selectedItems.rowKeys.length]);

    const onRow = useCallback((dzoData) => ({
        onClick: () => {
            const { dzoId } = dzoData;

            if (!select) {
                history.push(generatePath(`${matchPath}${DZO_PAGES.DZO_INFO}`, { dzoId }), { dzoData });
                return;
            }

            const rowKeys = !selectedItems.rowKeys.includes(dzoId)
                ? [...selectedItems.rowKeys, dzoId]
                : selectedItems.rowKeys.filter((selectedItemId) => selectedItemId !== dzoId);
            const rowValues = !selectedItems.rowValues.includes(dzoData)
                ? [...selectedItems.rowValues, dzoData]
                : selectedItems.rowValues.filter((selectedItem) => selectedItem.dzoId !== dzoId);
            setSelectedItems({ rowKeys, rowValues });
        },
    }), [select, selectedItems.rowKeys, selectedItems.rowValues, history, matchPath]);

    const handleSearch = useCallback((value = '') => {
        if (loadingTable) {
            return;
        }
        const nextParams = { ...params, filterText: value };
        sortDzoList(nextParams);
    }, [sortDzoList, loadingTable, params]);

    const changeSort = (sortBy) => {
        if (typeof sortBy !== 'string') {
            sortBy = '';
        }

        if (!sortBy && !params.sortBy) {
            return;
        }

        const sortParams = {
            ...params,
            sortBy,
            direction: !sortBy || params.direction === 'DESC' ? 'ASC' : 'DESC',
        };
        sortDzoList(sortParams);
    };

    const refreshTable = async () => {
        await loadDzoList(params);
        clearSelectedItems();
    };

    const toggleModal = () => {
        setIsModalView(!isModalView);
    };

    const buttons = [];

    if (select) {
        const selectedAll = dzoList.length === selectedItems.rowKeys.length;
        buttons.push({
            type: 'primary',
            label: selectedAll ? BUTTON_TEXT.CANCEL_ALL : BUTTON_TEXT.SELECT_ALL,
            onClick: selectAll,
            disabled: loadingTable,
        },
        {
            label: BUTTON_TEXT.CANCEL,
            onClick: clearSelectsRow,
            disabled: loadingTable,
        });
    } else {
        buttons.push(
            { type: 'primary', label: BUTTON_TEXT.ADD, onClick: onAddDzo, disabled: loadingTable },
            { label: BUTTON_TEXT.SELECT, onClick: onSelect, disabled: loadingTable },
        );
    }

    const searchInput = {
        placeholder: SEARCH_INPUT_PLACEHOLDER,
        onChange: handleSearch,
        value: params.filterText,
    };

    const sortingBy = {
        menuItems: DROPDOWN_SORT_MENU,
        onMenuItemClick: changeSort,
        withReset: true,
        sortBy: params.sortBy,
    };

    return (
        <>
            <div className={ styles.container }>
                <Header />
                <HeaderWithActions
                    buttons={ buttons }
                    showSorting
                    showSearchInput
                    searchInput={ searchInput }
                    sortingBy={ sortingBy }
                    classNameByInput={ styles.searchInput }
                    title={ DZO_TITLE }
                />
                <DzoListTable
                    loading={ loadingTable }
                    dzoList={ dzoList }
                    rowSelection={ select && rowSelection }
                    onRow={ onRow }
                />
                <div className={ styles.footer }>
                    { select && (
                        <div className={ styles.space }>
                            <div className={ styles.section }>
                                <span className={ styles.label }>
                                    { BUTTON_TEXT.CHOSEN } { selectedItems.rowKeys.length }
                                </span>
                            </div>
                            <Button
                                disabled={ !selectedItems.rowKeys.length }
                                danger
                                type='primary'
                                onClick={ toggleModal }
                            >
                                { BUTTON_TEXT.DELETE }
                            </Button>
                        </div>
                    ) }
                </div>
            </div>
            <TableDeleteModal
                listNameKey='dzoName'
                listIdForRemove={ selectedItems.rowKeys }
                sourceForRemove={ selectedItems.rowValues }
                modalClose={ toggleModal }
                deleteFunction={ deleteDzo }
                refreshTable={ refreshTable }
                modalSuccessTitle={ MODAL_SUCCESS_TITLE }
                modalTitle={ MODAL_TITLE }
                visible={ isModalView }
            />
        </>
    );
};

export default DzoPage;
