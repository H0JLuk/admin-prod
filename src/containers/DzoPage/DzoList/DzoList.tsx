import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { generatePath, useHistory } from 'react-router-dom';
import { Button } from 'antd';
import { deleteDzo, getAllDzoList, getDzoList } from '@apiServices/dzoService';
import DzoListTable from './Table/DzoListTable';
import Header from '@components/Header';
import TableDeleteModal from '@components/TableDeleteModal';
import HeaderWithActions, { ButtonProps } from '@components/HeaderWithActions';
import { DZO_PAGES } from '@constants/route';
import { requestsWithMinWait } from '@utils/utils';
import { DzoDto } from '@types';

import styles from './DzoList.module.css';

type IDzoPageProp = {
    matchPath: string;
};

type ISelectedItems = {
    rowKeys: number[];
    rowValues: DzoDto[];
};

type ISortByFieldKey = {
    NAME: keyof DzoDto;
};

const MODAL_TITLE = 'Вы уверены, что хотите удалить эти ДЗО?';
const MODAL_SUCCESS_TITLE = 'Результат удаления ДЗО';

const DZO_TITLE = 'Список ДЗО';

const DROPDOWN_SORT_MENU = [
    { name: 'dzoName', label: 'По названию' },
    { name: 'dzoId', label: 'По ID' },
    { name: 'dzoCode', label: 'По коду' },
];

const BUTTON_TEXT = {
    SAVE: 'Сохранить',
    ADD: 'Добавить',
    CANCEL: 'Отменить',
    SELECT: 'Выбрать',
    SELECT_ALL: 'Выбрать все',
    CANCEL_ALL: 'Отменить все',
    DELETE: 'Удалить',
    CHOSEN: 'Выбрано',
};

const SEARCH_INPUT_PLACEHOLDER = 'Поиск по названию дзо';

const sortByFieldKey: ISortByFieldKey = {
    NAME: 'dzoName',
};

const defaultSelected = { rowValues: [], rowKeys: [] };

const DzoPage: React.FC<IDzoPageProp> = ({ matchPath }) => {
    const history = useHistory();
    const [loadingTable, setLoadingTable] = useState(true);
    const [dzoList, setDzoList] = useState<DzoDto[]>([]);
    const copyDzoList = useRef<DzoDto[]>([]);
    const [select, setSelect] = useState(false);
    const [selectedItems, setSelectedItems] = useState<ISelectedItems>(defaultSelected);
    const dzoCodes = useRef<string[]>([]);

    const onAddDzo = () => history.push(generatePath(`${matchPath}${DZO_PAGES.ADD_DZO}`), { dzoCodes: dzoCodes.current });

    const loadDzoList = useCallback(async () => {
        setLoadingTable(true);
        try {
            const requests = Promise.all([getDzoList(), getAllDzoList()]);
            const [
                { dzoDtoList = [] },
                { dzoDtoList: allDzo = [] },
            ] = await requestsWithMinWait(requests, 0);
            dzoCodes.current = allDzo.map(({ dzoCode }) => dzoCode);
            copyDzoList.current = dzoDtoList;
            setDzoList(dzoDtoList);
            clearSelectedItems();
        } catch (e) {
            console.warn(e);
        }
        setLoadingTable(false);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    useEffect(() => {
        loadDzoList();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [loadDzoList]);

    const clearSelectedItems = () => {
        setSelectedItems(defaultSelected);
    };

    const clearSelectsRow = () => {
        setSelect(false);
        clearSelectedItems();
    };

    const updateSelected = useCallback((rowKeys, rowValues) => {
        setSelectedItems({ rowKeys, rowValues });
    }, []);

    const onSelect = useCallback(() => setSelect(true), []);

    const rowSelection = useMemo(() => ({
        selectedRowKeys: selectedItems.rowKeys,
        onChange: updateSelected,
    }), [selectedItems, updateSelected]);

    const selectAll = useCallback(() => {
        if (dzoList.length !== selectedItems.rowKeys.length) {
            setSelectedItems({ rowKeys: dzoList.map(({ dzoId }) => dzoId), rowValues: dzoList });
            return;
        }
        clearSelectedItems();
    }, [dzoList, selectedItems.rowKeys.length]);

    const onRow = useCallback((dzoData) => ({
        onClick: () => {
            const { dzoId } = dzoData;

            if (!select) {
                history.push(generatePath(`${matchPath}${DZO_PAGES.DZO_INFO}`, { dzoId }), { dzoData });
                return;
            }

            const rowKeys = !selectedItems.rowKeys.includes(dzoId as never)
                ? [...selectedItems.rowKeys, dzoId]
                : selectedItems.rowKeys.filter((selectedItemId) => selectedItemId !== dzoId);
            const rowValues = !selectedItems.rowValues.includes(dzoData as never)
                ? [...selectedItems.rowValues, dzoData]
                : selectedItems.rowValues.filter((selectedItem) => selectedItem.dzoId !== dzoId);
            setSelectedItems({ rowKeys, rowValues });
        },
    }), [select, selectedItems.rowKeys, selectedItems.rowValues, history, matchPath]);

    const refreshTable = () => {
        loadDzoList();
        clearSelectedItems();
    };

    const buttons: ButtonProps[] = [];

    if (select) {
        const selectedAll = dzoList.length === selectedItems.rowKeys.length;
        buttons.push({
            type: 'primary',
            label: selectedAll ? BUTTON_TEXT.CANCEL_ALL : BUTTON_TEXT.SELECT_ALL,
            onClick: selectAll,
            disabled: loadingTable,
        },
        {
            label: BUTTON_TEXT.CANCEL,
            onClick: clearSelectsRow,
            disabled: loadingTable,
        });
    } else {
        buttons.push(
            { type: 'primary', label: BUTTON_TEXT.ADD, onClick: onAddDzo, disabled: loadingTable },
            { label: BUTTON_TEXT.SELECT, onClick: onSelect, disabled: loadingTable },
        );
    }

    const searchAndSortParams = {
        setDataList: setDzoList,
        copyDataList: copyDzoList.current,
        matchPath,
        inputPlaceholder: SEARCH_INPUT_PLACEHOLDER,
        sortByFieldKey: sortByFieldKey.NAME,
        menuItems: DROPDOWN_SORT_MENU,
        onChangeInput: clearSelectedItems,
    };

    return (
        <>
            <div className={styles.container}>
                <Header />
                {<HeaderWithActions<DzoDto>
                    title={DZO_TITLE}
                    buttons={buttons}
                    {...searchAndSortParams}
                />}
                <DzoListTable
                    loading={loadingTable}
                    dzoList={dzoList}
                    rowSelection={select ? rowSelection : undefined}
                    onRow={onRow}
                />
                {select && (
                    <div className={styles.footer}>
                        <div className={styles.space}>
                            <div className={styles.section}>
                                <span className={styles.label}>
                                    {BUTTON_TEXT.CHOSEN} {selectedItems.rowKeys.length}
                                </span>
                            </div>
                            <TableDeleteModal<DzoDto>
                                listNameKey="dzoName"
                                listIdForRemove={selectedItems.rowKeys}
                                sourceForRemove={selectedItems.rowValues}
                                deleteFunction={deleteDzo}
                                refreshTable={refreshTable}
                                modalSuccessTitle={MODAL_SUCCESS_TITLE}
                                modalTitle={MODAL_TITLE}
                            >
                                <Button
                                    type="primary"
                                    disabled={!selectedItems.rowKeys.length}
                                    danger
                                >
                                    {BUTTON_TEXT.DELETE}
                                </Button>
                            </TableDeleteModal>
                        </div>
                    </div>
                )}
            </div>
        </>
    );
};

export default DzoPage;
