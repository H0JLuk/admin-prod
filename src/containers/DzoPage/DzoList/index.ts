import DzoList from './DzoList';

export default DzoList;
