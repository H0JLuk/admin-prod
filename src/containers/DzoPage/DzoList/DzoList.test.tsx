import React from 'react';
import { generatePath } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { deleteDzo, getAllDzoList, getDzoList } from '@apiServices/dzoService';
import { BUTTON_TEXT } from '@constants/common';
import { requestsWithMinWait } from '../../../utils/utils';
import DzoList from './DzoList';
import { dzoListTestData } from '../../../../__tests__/constants';
import { sleep } from '../../../setupTests';

type mockTableDeleteModalType = {
    listIdForRemove: number[];
    refreshTable: () => void;
    deleteFunction: () => void;
};

jest.mock('@components/TableDeleteModal',
    () =>
        ({ listIdForRemove, refreshTable, deleteFunction }: mockTableDeleteModalType) => (
            <div data-testid="checkboxes">
                <span data-testid="listIdForRemove">{JSON.stringify(listIdForRemove)}</span>
                <button data-testid="refreshTable" onClick={refreshTable}>Хорошо</button>
                <button data-testid="deleteFunction" onClick={deleteFunction}>Удалить</button>
            </div>
        ));

const props = {
    matchPath: 'matchPath',
};
const mockHistoryPush = jest.fn();
const mockHistoryReplace = jest.fn();

jest.mock('../../../api/services/dzoService', () => ({
    deleteDzo: jest.fn(),
    getDzoList: jest.fn(),
    getAllDzoList: jest.fn(),
}));
jest.mock('../../../utils/utils', () => ({
    requestsWithMinWait: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: mockHistoryPush,
        replace: mockHistoryReplace,
        location: { search: '?sortBy=&direction=ASC&filterText=' },
    }),
    generatePath: jest.fn(),
}));

describe('<DzoFrom /> test', () => {
    beforeEach(() => {
        (requestsWithMinWait as jest.Mock).mockResolvedValue([
            { dzoDtoList: dzoListTestData },
            { dzoDtoList: dzoListTestData },
        ]);
        (generatePath as jest.Mock).mockReturnValue(props.matchPath);
    });

    it('should match snapshot', async () => {
        const { asFragment } = render(<DzoList {...props} />);

        await screen.findByText(dzoListTestData[0].dzoName);

        expect(asFragment()).toMatchSnapshot('after loading data');
        expect(getAllDzoList).toBeCalledTimes(1);
        expect(getDzoList).toBeCalledTimes(1);
    });

    it('should handle fetch dzo error', async () => {
        const warnMessage = 'test warn';
        (requestsWithMinWait as jest.Mock).mockRejectedValue(warnMessage);
        const spyOnConsole = jest
            .spyOn(global.console, 'warn')
            .mockImplementation(() => '');

        render(<DzoList {...props} />);
        await sleep();

        expect(spyOnConsole).toBeCalledTimes(1);
        expect(spyOnConsole).toBeCalledWith(warnMessage);
    });

    it('should call history push on add button', async () => {
        render(<DzoList {...props} />);

        await userEvent.click(await screen.findByText(BUTTON_TEXT.ADD));

        const dzoCodes = dzoListTestData.map(({ dzoCode }) => dzoCode);
        expect(mockHistoryPush).toBeCalledWith(props.matchPath, { dzoCodes });
    });

    it('should call history push on dzo row click', async () => {
        render(<DzoList {...props} />);

        userEvent.click(await screen.findByText(dzoListTestData[0].dzoName)),

        expect(mockHistoryPush).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledWith(props.matchPath, { dzoData: dzoListTestData[0] });
    });

    it('should select and unselect items at rows click', async () => {
        render(<DzoList {...props} />);

        userEvent.click(await screen.findByRole('button', { name: BUTTON_TEXT.SELECT }));
        expect(screen.getByText('Выбрано 0')).toBeInTheDocument();

        const firstItem = screen.getByText(dzoListTestData[0].dzoName);
        userEvent.click(firstItem);
        expect(screen.getByText('Выбрано 1')).toBeInTheDocument();

        userEvent.click(firstItem);
        expect(screen.getByText('Выбрано 0')).toBeInTheDocument();
    });

    it('should select all items then clear it', async () => {
        render(<DzoList {...props} />);

        userEvent.click(await screen.findByRole('button', { name: BUTTON_TEXT.SELECT }));

        expect(screen.getByText('Выбрано 0')).toBeInTheDocument();

        userEvent.click(screen.getByText(BUTTON_TEXT.SELECT_ALL));

        expect(screen.getByText('Выбрано 3')).toBeInTheDocument();

        userEvent.click(screen.getByText(BUTTON_TEXT.CANCEL_ALL));

        expect(screen.getByText('Выбрано 0')).toBeInTheDocument();

        userEvent.click(screen.getByRole('button', { name: BUTTON_TEXT.CANCEL }));

        expect(screen.getByRole('button', { name: BUTTON_TEXT.SELECT })).toBeInTheDocument();
    });

    it('should select item at checkbox click', async () => {
        render(<DzoList {...props} />);

        await act(async () =>
            userEvent.click(await screen.findByRole('button', { name: BUTTON_TEXT.SELECT })),
        );

        const rowName = new RegExp(dzoListTestData[0].dzoName);
        const row = screen.getByRole('row', { name: rowName });
        const checkbox = within(row).getByRole('checkbox');
        userEvent.click(checkbox);

        expect(screen.getByText('Выбрано 1')).toBeInTheDocument();
    });

    it('should delete any dzo rows on submit modal window', async () => {
        render(<DzoList {...props} />);

        await act(async () =>
            userEvent.click(await screen.findByRole('button', { name: BUTTON_TEXT.SELECT })),
        );

        userEvent.click(screen.getByText(dzoListTestData[0].dzoName));
        userEvent.click(screen.getByText(dzoListTestData[1].dzoName));

        expect(screen.getByText('Выбрано 2')).toBeInTheDocument();

        expect(deleteDzo).toBeCalledTimes(0);
        act(() =>
            userEvent.click(screen.getByRole('button', { name: BUTTON_TEXT.DELETE })),
        );
        expect(deleteDzo).toBeCalledTimes(1);
        act(() =>
            userEvent.click(screen.getByTestId('deleteFunction')),
        );
        expect(deleteDzo).toBeCalledTimes(2);

        expect(getAllDzoList).toBeCalledTimes(1);
        expect(getDzoList).toBeCalledTimes(1);
        await act(async () =>
            userEvent.click(await screen.findByTestId('refreshTable')),
        );
        expect(getAllDzoList).toBeCalledTimes(2);
        expect(getDzoList).toBeCalledTimes(2);
    });
});
