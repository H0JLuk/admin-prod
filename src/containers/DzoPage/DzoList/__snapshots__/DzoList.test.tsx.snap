// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DzoFrom /> test should match snapshot: after loading data 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="backButton"
      >
        <span
          aria-label="left"
          class="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
          Назад
      </div>
      <div
        class="avatar"
      >
        <span
          aria-label="user"
          class="anticon anticon-user ant-dropdown-trigger dropdown"
          role="img"
          style="font-size: 22px; color: rgb(255, 255, 255); background-color: rgb(193, 200, 209); border-radius: 50px; width: 32px; height: 32px; padding: 3px;"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="headerTitle"
      >
        Список ДЗО
      </div>
      <div
        class="actions"
      >
        <div
          class="buttons"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Добавить
            </span>
          </button>
          <button
            class="ant-btn"
            type="button"
          >
            <span>
              Выбрать
            </span>
          </button>
          <span
            class="ant-input-affix-wrapper"
          >
            <span
              class="ant-input-prefix"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </span>
            <input
              class="ant-input"
              placeholder="Поиск по названию дзо"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                role="button"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <button
          class="ant-btn ant-dropdown-trigger dropdownButton"
          type="button"
        >
          <span>
            Сортировать 
          </span>
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
            style="font-size: 14px; color: rgb(195, 201, 208);"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="tableWrapper"
    >
      <div
        class="ant-table-wrapper table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 45%;"
                      />
                      <col
                        style="width: 40%;"
                      />
                      <col
                        style="width: 15%;"
                      />
                    </colgroup>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Название
                          </div>
                          <div
                            class="tableCell"
                          >
                            Беру!
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            ID
                          </div>
                          <div
                            class="tableCell"
                          >
                            0
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Код
                          </div>
                          <div
                            class="tableCell"
                          >
                            beru_code
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="3"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Название
                          </div>
                          <div
                            class="tableCell"
                          >
                            Сбермаркет
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            ID
                          </div>
                          <div
                            class="tableCell"
                          >
                            3
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Код
                          </div>
                          <div
                            class="tableCell"
                          >
                            sbermarket_old
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="70"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Название
                          </div>
                          <div
                            class="tableCell"
                          >
                            СберМобайл
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            ID
                          </div>
                          <div
                            class="tableCell"
                          >
                            70
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="tableTitle"
                          >
                            Код
                          </div>
                          <div
                            class="tableCell"
                          >
                            sbermobile
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
